packages/app/src/cli/commands/app/function/run.ts

```diff
@@ -42,10 +42,43 @@ export default class FunctionRun extends Command {
       path: flags.path,
       userProvidedConfigName: flags.config,
       callback: async (_app, ourFunction) => {
+        let functionExport = '_start'
+
+        if (flags.export !== undefined) {
+          this.log('Using export ${flags.export} from the --export flag.')
+          functionExport = flags.export
+        } else if (
+          ourFunction.configuration.targeting !== undefined &&
+          ourFunction.configuration.targeting.length > 0
+        ) {
+          const targeting = ourFunction.configuration.targeting
+
+          if (targeting.length > 1 && isTTY()) {
```

Comment:
If there are > 1 targets and this isn't interactive, wouldn't it not fall into the `else if`, which means the export is `_start`. Based on the PR description, I think we want to choose the first export in that case.